src/transaction/mod.rs

```diff
@@ -64,7 +79,11 @@ impl Transaction {
             Transaction::L1Handler(tx) => tx.execute(state, block_context, remaining_gas),
         }
     }
-
+    /// It creates a new transaction structure modificating the skip flags. It is meant to be used only to run a simulation
+    ///## Parameters:
+    ///- skip_validate: the transaction will not be verificatedi.
+    ///- skip_execute: the transaction will not be executed in the cairo vmi.
+    ///- skip_fee_transfer: the transaction will not consume gas as it will skip fee_transfer.
```

Comment:
Isn't necessary for users to know about the internals. They doesn't need to know about the `fee_trasnfer` function.
```suggestion
    ///- skip_fee_transfer: the transaction will not pay the fee.
```